### PR TITLE
feat(dialog) new native InputDialog

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

File: react/features/base/dialog/components/native/InputDialog.js [hunk 1]
```diff
@@ -142,7 +142,7 @@ class InputDialog extends BaseDialog<Props, State> {
     _onChangeText(fieldValue) {
 
         if (this.props.validateInput
-                && !this.props.validateInput(fieldValue)) {
+            && !this.props.validateInput(fieldValue)) {
```
Comment (saghul) [inline react/features/base/dialog/components/native/InputDialog.js:145]: unrelated change

File: react/features/base/dialog/components/native/InputDialog.js [hunk 2]
```diff
@@ -79,52 +82,49 @@ class InputDialog extends BaseDialog<Props, State> {
...
+                    onBackdropPress = { this._onCancel }
+                    visible = { true }>
+                    {
+                        titleKey && (
+                            <Dialog.Title>
+                                { t(titleKey) }
+                            </Dialog.Title>
+                        )
+                    }
+                    { children }
```
[the remark this answers is not in the document]
Comment (saghul) [reply react/features/base/dialog/components/native/InputDialog.js:109]: I see we pass the description as children. Shouldn't we take the description as a prop instead? Sounds like a pretty standard thing to want I think.

File: react/features/shared-video/components/native/SharedVideoDialog.js [hunk 3]
```diff
@@ -56,15 +56,15 @@ class SharedVideoDialog extends AbstractSharedVideoDialog<*> {
...
-                contentKey = 'dialog.shareVideoTitle'
                 messageKey = { error ? 'dialog.sharedVideoDialogError' : undefined }
                 onSubmit = { this._onSubmitValue }
                 textInputProps = {{
                     autoCapitalize: 'none',
                     autoCorrect: false,
                     placeholder: t('dialog.sharedVideoLinkPlaceholder'),
-                    placeholderTextColor: ColorPalette.lightGrey
-                }} />
+                    placeholderTextColor: BaseTheme.palette.text03
```
Comment (saghul) [inline react/features/shared-video/components/native/SharedVideoDialog.js:65]: I think we better leave this to the react-native-dialog package, it already knows what the right color is: https://github.com/mmazzarolo/react-native-dialog/blob/5a69064a1fe5c99f9866ddadddb7c197c2728acc/src/Input.tsx#L43

**Review by saghul — CHANGES_REQUESTED** [review]
Some minor comments, PTAL.

File: react/features/base/dialog/components/native/InputDialog.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Calinteodor) [reply react/features/base/dialog/components/native/InputDialog.js:109]: Yes. I wanted to apply the same logic that I applied for ConfirmDialog, but yeah, here we only use strings and it's not necessary.

[the remark this answers is not in the document]
Comment (saghul) [reply react/features/base/dialog/components/native/InputDialog.js:109]: Let's go simple, since we have no other use case at the moment.

**Review by saghul — APPROVED** [review]
🚀 🚀 🚀 🚀
